modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackManager.kt

```diff
@@ -1503,19 +1503,29 @@ open class PlaybackManager @Inject constructor(
         if (episode.isFinished) {
             episodeManager.markAsNotPlayed(episode)
         }
-        // check we have the latest episode url in the background
-        if (episode is PodcastEpisode) {
-            updateEpisodeUrl(episode)
-        }
 
-        if (episode is UserEpisode && episode.serverStatus == UserEpisodeServerStatus.UPLOADED) {
-            try {
-                val playbackUrl = userEpisodeManager.getPlaybackUrl(episode).await()
-                episode.downloadUrl = playbackUrl
-            } catch (e: Exception) {
-                onPlayerError(PlayerEvent.PlayerError("Could not load cloud file ${e.message}"))
-                removeEpisode(episode, source = sourceView)
-                return
+        // make sure we have the latest episode url
+        when (episode) {
+            is PodcastEpisode -> {
+                if (!episode.isDownloaded) {
+                    val newDownloadUrl = episodeManager.updateDownloadUrl(episode)
+                    if (newDownloadUrl != null && newDownloadUrl != episode.downloadUrl) {
+                        Timber.i("Updating url for PodcastEpisode playback")
+                        episode.downloadUrl = newDownloadUrl
+                    }
+                }
+            }
+            is UserEpisode -> {
+                if (episode.serverStatus == UserEpisodeServerStatus.UPLOADED) {
+                    try {
+                        val newDownloadUrl = userEpisodeManager.getPlaybackUrl(episode).await()
+                        episode.downloadUrl = newDownloadUrl
+                    } catch (e: Exception) {
+                        onPlayerError(PlayerEvent.PlayerError("Could not load cloud file ${e.message}"))
+                        removeEpisode(episode, source = sourceView)
+                        return
+                    }
+                }
             }
```

Comment:
Apart from moving this handling for user episodes into the when statement, the actual handling for use episode's should be unchanged. I just thought using a when statement here was nice since it helps show highlight that we're updating the URLs here differently for user episodes and podcast episodes.